### PR TITLE
Add positional placeholder normalization to PrintfLikeIntegrityChecker

### PR DESCRIPTION
**Pull request by vnpins**
## Issue
The `PRINTF_LIKE` integrity checker was rejecting valid translations where:
- Source strings use non-positional placeholders (e.g., `%s`, `%d`)
- Target translations use positional placeholders (e.g., `%1$s`, `%1$d`)

This commonly occurs with Smartling, which automatically adds position specifiers to avoid ambiguity in translations.

### Example Rejection
- Source: `"Select %s answers to proceed"`
- Target: `"Kies %1$s antwoorde om voort te gaan"` ❌ REJECTED
- Reason: `%s` ≠ `%1$s` (exact string match failed)

## Solution
Modified `PrintfLikeIntegrityChecker` to normalize both source and target strings before placeholder comparison:
- Non-positional placeholders → converted to positional format
- `%s` → `%1$s`, `%d` → `%1$d`, `%.1f` → `%1$.1f`, etc.
- Placeholders that already have positional specifiers are left unchanged
- Comparison happens after normalization, allowing semantic equivalence

### After Fix
- Source: `"Select %s answers"` → normalized to `"Select %1$s answers"`
- Target: `"Kies %1$s antwoorde"` → already `"Kies %1$s antwoorde"`
- Result: ✓ PASS

## Changes
1. **PrintfLikeIntegrityChecker.java**
   - Added placeholder normalization before extraction
   - Converts non-positional placeholders to positional format automatically
   - Made normalization the default behavior for all printf-like checks

2. **PrintfLikeAddParameterSpecifierIntegrityChecker.java**
   - Added documentation noting equivalence with base class
   - Kept for backward compatibility

3. **PrintfLikeIntegrityCheckerTest.java**
   - Added 4 positive test cases (non-positional ↔ positional matching)
   - Added 4 negative test cases (verify invalid cases still fail)
   - All 16 existing tests continue to pass

4. **Documentation**
   - Updated integrity checker guide with examples
   - Added edge case documentation

## Testing
- ✅ All 51 unit tests pass (29 PrintfLikeIntegrityChecker + 22 PrintfLikeAddParameterSpecifierIntegrityChecker)
- ✅ 8 new tests added (4 positive, 4 negative)
- ✅ All existing tests continue to pass (regression check)
- ✅ No regressions in other integrity checkers

## Edge Cases
When source has multiple identical non-positional placeholders (e.g., `"Copy %s to %s"`), they normalize to the same placeholder (`%1$s %1$s`). If target uses distinct positions (`%1$s %2$s`), the check fails. This is expected and indicates the source should use positional format. Note that repos using `SmartlingSourceStringConverter` already handle this by assigning sequential positions before sending to Smartling.

### CODE REVIEW

File: webapp/src/test/java/com/box/l10n/mojito/service/assetintegritychecker/integritychecker/PrintfLikeIntegrityCheckerTest.java [hunk 1]
```diff
@@ -285,4 +285,112 @@ public void testIncorrectlyModifiedSpecifier() throws PrintfLikeIntegrityChecker
...
+    String source = "There are %s files and %d folders";
+    String target = "Il y a %1$s fichiers";
+
+    try {
+      checker.check(source, target);
+      fail("PrintfLikeIntegrityCheckerException must be thrown");
+    } catch (PrintfLikeIntegrityCheckerException e) {
+      assertEquals(e.getMessage(), "Placeholders in source and target are different");
+    }
+  }
```
Comment (maallen) [inline webapp/src/test/java/com/box/l10n/mojito/service/assetintegritychecker/integritychecker/PrintfLikeIntegrityCheckerTest.java:395]: Can you add a test where there's a raw `%` in the text? e.g. `95% of strings`

**Review by maallen — APPROVED** [review]

**Review by mattwilshire — APPROVED** [review]

File: webapp/src/test/java/com/box/l10n/mojito/service/assetintegritychecker/integritychecker/PrintfLikeIntegrityCheckerTest.java (hunk 1 above)
Comment (vnpins) [reply webapp/src/test/java/com/box/l10n/mojito/service/assetintegritychecker/integritychecker/PrintfLikeIntegrityCheckerTest.java:395]: Done!
